### PR TITLE
feat: Nested source directories

### PR DESCRIPTION
**Pull request by tbarlow12**
Adds capability of using all nested assets within a source directory.

Addresses #839

### CODE REVIEW

File: src/electron/providers/storage/localFileSystem.ts [hunk 1]
```diff
@@ -137,9 +147,12 @@ export default class LocalFileSystem implements IStorageProvider {
         });
     }
 
-    public async getAssets(sourceConnectionFolderPath?: string, relativePath: boolean = false): Promise<IAsset[]> {
-        return (await this.listFiles(path.normalize(sourceConnectionFolderPath)))
-            .map((filePath) => AssetService.createAssetFromFilePath(
+    public async getAssets(
+            sourceConnectionFolderPath?: string,
+            relativePath: boolean = false,
+            recursive: boolean = true): Promise<IAsset[]> {
```
Comment (wbreza) [inline src/electron/providers/storage/localFileSystem.ts:153]: Any reason to not always have `recursive` set??

File: src/electron/providers/storage/localFileSystem.ts [hunk 2]
```diff
@@ -93,8 +93,18 @@ export default class LocalFileSystem implements IStorageProvider {
         });
     }
 
-    public listFiles(folderPath: string): Promise<string[]> {
-        return this.listItems(path.normalize(folderPath), (stats) => !stats.isDirectory());
+    public async listFiles(folderPath: string, ext?: string, recursive: boolean = false): Promise<string[]> {
```
Comment (wbreza) [inline src/electron/providers/storage/localFileSystem.ts:96]: Same, any reason not do always have this as true?

**Review by wbreza — APPROVED** [review]
Looks great - Added some feedback on whether we should just always default to `recursive: true`?

File: src/electron/providers/storage/localFileSystem.ts (hunk 2 above)
Comment (tbarlow12) [reply src/electron/providers/storage/localFileSystem.ts:96]: Wasn't sure. Was leaving it open for flexibility, but I'll remove that param

**Review by jding05 — APPROVED** [review]
LGTM

**Pull request merged**